src/rest/DataUnionEndpoints.js

```diff
@@ -39,163 +184,222 @@ function throwIfBadAddress(address, variableDescription) {
 
-/**
- * Await this function when you want to make sure a data union is deployed and ready to use
- * @param {EthereumAddress} dataUnionContractAddress
- * @param {Number} pollingIntervalMs (optional, default: 1000) ask server if data union is ready
- * @param {Number} retryTimeoutMs (optional, default: 60000) give up sending more retries
- * @return {Promise} resolves when data union server is ready to operate the data union (or fails with HTTP error)
- */
-export async function dataUnionIsReady(dataUnionContractAddress, pollingIntervalMs = 1000, retryTimeoutMs = 60000) {
-    let stats = await get(this, dataUnionContractAddress, '/stats')
-    const startTime = Date.now()
-    while (stats.error && Date.now() < startTime + retryTimeoutMs && (!stats.dataUnion || stats.dataUnion.state !== 'failed')) {
-        debug(`Waiting for data union ${dataUnionContractAddress} to start. Status: ${JSON.stringify(stats)}`)
-        await sleep(pollingIntervalMs) // eslint-disable-line no-await-in-loop
-        stats = await get(this, dataUnionContractAddress, '/stats') // eslint-disable-line no-await-in-loop
+    if (await mainnetProvider.getCode(duMainnetAddress) !== '0x') {
+        throw new Error(`Mainnet data union "${duName}" contract ${duMainnetAddress} already exists!`)
     }
-    if (stats.error) {
-        throw new Error(`Data Union failed to start, retried for ${retryTimeoutMs} ms. Status: ${JSON.stringify(stats)}`)
+
+    const factoryMainnetAddress = throwIfBadAddress(
+        options.factoryMainnetAddress || this.options.factoryMainnetAddress,
+        'StreamrClient.options.factoryMainnetAddress'
+    )
+    if (await mainnetProvider.getCode(factoryMainnetAddress) === '0x') {
+        throw new Error(`Data union factory contract not found at ${factoryMainnetAddress}, check StreamrClient.options.factoryMainnetAddress!`)
     }
+    const factoryMainnet = new Contract(factoryMainnetAddress, factoryMainnetABI, mainnetWallet)
+
+    // function deployNewDataUnion(address owner, uint256 adminFeeFraction, address[] agents, string duName)
+    const tx = await factoryMainnet.deployNewDataUnion(
+        ownerAddress,
+        adminFeeBN,
+        agentAddressList,
+        duName,
+    )
+    const tr = await tx.wait()
+
+    log(`Data Union "${duName}" (mainnet: ${duMainnetAddress}, sidechain: ${duSidechainAddress}) deployed to mainnet, waiting for side-chain...`)
+    await until(
+        async () => await sidechainWallet.provider.getCode(duSidechainAddress) !== '0x',
+        sidechainRetryTimeoutMs,
+        sidechainPollingIntervalMs
+    )
+
+    const dataUnion = new Contract(duMainnetAddress, dataUnionMainnetABI, mainnetWallet)
+    dataUnion.deployTxReceipt = tr
+    dataUnion.sidechain = new Contract(duSidechainAddress, dataUnionSidechainABI, sidechainWallet)
+    return dataUnion
+}
+
+export async function getDataUnionContract(options = {}) {
+    const ret = getMainnetContract(this, options)
```

Comment:
see comment about `getMainnetContract`, need to await here:

```js
const ret = await getMainnetContract(this, options)
```

Reply:
changed it so that await isn't needed